Generic Challenge.playground/Contents.swift

```diff
@@ -0,0 +1,56 @@
+import UIKit
+
+struct CountedSet<Element: Hashable> {
+    
+    private var dictionary = [Element: Int] ()
+    var count: Int {
+        return dictionary.count
+    }
+    
+    var isEmpty: Bool {
+        return count == 0
+    }
+    
+    mutating func insert(element: Element) {
+        if let elementCount = dictionary[element] {
+            dictionary[element] = elementCount + 1
+        } else {
+            dictionary[element] = 1
+        }
+    }
```

Comment:
```suggestion
    mutating func insert(element: Element) {
        dictionary[element, default: 0] += 1
    }
```
This is a short hand for the exact same logic you had in your insert, there is nothing wrong with how you did it, just thought I would share.